app/Http/Controllers/Admin/NewsController.php

```diff
@@ -61,4 +62,45 @@ public function store(CreateNewsRequest $request)
             return redirect()->route('news.create');
         }
     }
+
+    /**
+     * Display form edit a News.
+     *
+     * @param string $slug of News
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($slug)
+    {
+        $columns = [
+                    'id',
+                    'title',
+                    'content',
+                    'category_id'
+        ];
+        $news = News::select($columns)
+                    ->where('slug', $slug)
+                    ->first();
```

Comment:
@at-minhnguyen i think if use `Slug`for find this ok and @AT-DuocNguyen need check :  

```
if (is_null($news)) { 
  throw new ModelNotFoundException; 
}
```
**(use \Illuminate\Database\Eloquent\ModelNotFoundException;)**